app/components/rate-goal/animations/cactus.jsx

```diff
@@ -0,0 +1,128 @@
+import React from 'react';
+import GSAP from 'react-gsap-enhancer';
+import { TweenMax, TimelineMax, } from 'gsap';
+
+const createAnimation = ({target, options, }) => {
+
+  const bodyFrames = [
+    { bottom: '50px', height: '0px', },
+    { bottom: '54px', height: '50px', },
+    { bottom: '66px', height: '50px', },
+    { bottom: '70px', height: '60px', },
+    { bottom: '75px', height: '65px', },
+    { bottom: '75px', height: '70px', },
+    { bottom: '80px', height: '70px', },
+    { bottom: '85px', height: '70px', },
+    { bottom: '90px', height: '72px', },
+    { bottom: '95px', height: '77px', },
+    { bottom: '97px', height: '81px', },
+  ];
+
+  const armFrames = [
+    { visibility: 'hidden' },
+    { visibility: 'hidden' },
+    { visibility: 'hidden' },
+    { visibility: 'hidden' },
+    { visibility: 'visible' },
+    { visibility: 'visible' },
+    { visibility: 'visible' },
+    { visibility: 'visible' },
+    { visibility: 'visible' },
+    { visibility: 'visible' },
+    { visibility: 'visible' },
+  ];
+
+  const vase = target.find({ id: 'vase' });
+  const cactus3 = target.find({ id: 'cactus3' });
+  const arms = target.find({ id: 'arms' });
+
+  let t1 = new TimelineMax();
+
+    const sequence = options.score > options.previousScore
+      ? bodyFrames.slice(options.previousScore, options.score + 1)
+      : bodyFrames.slice(options.score, options.previousScore + 1).reverse();
+
+      const armSequence = options.score > options.previousScore
+        ? armFrames.slice(options.previousScore, options.score + 1)
+        : armFrames.slice(options.score, options.previousScore + 1).reverse();
+
+    sequence.forEach((frame, index) => {
+      t1.to(cactus3, 0.3, frame, )
+        .to(arms, 0.5, armSequence[index], "-=0.5")
+    });
+
+    t1.eventCallback("onComplete", options.setPreviousScore());
+
+    return t1;
+}
+
+class Cactus extends React.Component {
+
+  constructor(props) {
+    super(props)
+    this.state = {
+      previousScore: this.props.previousScore,
+      score: this.props.score,
+    };
+  };
+
+  componentWillMount() {
+      this.timerID = setInterval(
+        () => this.tick(),
+        100);
+  }
+
+  componentWillUnmount() {
+    clearInterval(this.timerID);
+  }
+
+  tick() {
+    this.setState({
+      previousScore: this.props.previousScore,
+      score: this.props.score,
+    });
+    const options = {
+      ...this.state,
+      setPreviousScore: this.props.setPreviousScore,
+    };
+
+    if (options.score === options.previousScore) {
+      return;
+    }
+    this.animation = this.addAnimation(createAnimation, options);
```

Comment:
I am not sure `setPreviousScore` should be in options, as it is a function this is kinda confusing.. it is not optional, `addAnimation` will crash without it. I would make it a normal argument